src/authConfig.ts

```diff
@@ -16,37 +19,81 @@ declare global {
     }
 }
 
-export const createMsalConfig = (clientId: string, authority: string, redirectUri: string): Configuration => {
-    return {
-        auth: {
-            // the following two values are changed using the Environment Provider in the layout.tsx
-            clientId,
-            authority, // This is a URL (e.g. https://login.microsoftonline.com/{your tenant ID})
-            redirectUri,
+const keycloakEnabled = process.env.KEYCLOAK_ENABLED?.toLowerCase() === 'true'.toLowerCase();
+
+export const authOptions: AuthOptions = {
+    providers: [
+        ...(keycloakEnabled ? [
+            KeycloakProvider({
+                clientId: process.env.KEYCLOAK_CLIENT_ID ? process.env.KEYCLOAK_CLIENT_ID : '',
+                clientSecret: 'process.env.KEYCLOAK_CLIENT_SECRET',
+                issuer: process.env.KEYCLOAK_ISSUER
+            })
+        ] : [
+            AzureADProvider({
+                clientId: process.env.AD_CLIENT_ID ? process.env.AD_CLIENT_ID : '',
+                clientSecret: process.env.AD_SECRET_VALUE ? process.env.AD_SECRET_VALUE : '',
```

Comment:
I think it should also work like this:
```suggestion
                clientSecret: process.env.AD_SECRET_VALUE ?? '',
```